### PR TITLE
add MobileNetV3 as an Encoder

### PR DESCRIPTION
**Pull request by markson14**
[PR] MobileNetV3 Encoder Added !!
1. Modifying MobileNetV3 as an Encoder
2. Using "mode" and "width_mult" to manipulate out_channels. Encoder can be easily extended.

### CODE REVIEW

File: segmentation_models_pytorch/encoders/mobilenet.py [hunk 1]
```diff
@@ -25,10 +25,277 @@
...
+        if mode == "small":
+            self._out_channels = (3*width_mult, 16*width_mult, 16*width_mult,
+                                  24*width_mult, 48*width_mult, 96*width_mult)
+            self._out_channels = tuple(map(make_divisible, self._out_channels))
+        elif mode == "large":
+            self._out_channels = (3*width_mult, 16*width_mult, 24*width_mult,
+                                  40*width_mult, 112*width_mult, 160*width_mult)
+            self._out_channels = tuple(map(make_divisible, self._out_channels))
+        else:
+            raise NameError
```
Comment (qubvel) [inline segmentation_models_pytorch/encoders/mobilenet.py:262]: raise ValueError("MobileNetV3 mode should be `small` or `large`, got {}".format(mode))

File: segmentation_models_pytorch/encoders/mobilenet.py [hunk 2]
```diff
@@ -25,10 +25,277 @@
...
+        self._in_channels = 3
+
+        del self.classifier
```
Comment (qubvel) [inline segmentation_models_pytorch/encoders/mobilenet.py:265]: self.classifier deleted but `load_state_dict` is not overwritten, is weights loading ok with that?

**Comment by qubvel** [comment]
Hi, thanks a lot for PR, we need some more actions to merge it.

Add information about Encoder in following places:

1) README.md modify number of encoder in start description
2) README.md add encoder names to table
3) Add information about encoders to docs/encoders.rst

You can use misc/generate_table.py script to get number of params for encoders.

**Comment by markson14** [comment]
> Hi, thanks a lot for PR, we need some more actions to merge it.
> 
> Add information about Encoder in following places:
> 
> 1. README.md modify number of encoder in start description
> 2. README.md add encoder names to table
> 3. Add information about encoders to docs/encoders.rst
> 
> You can use misc/generate_table.py script to get number of params for encoders.

Sure, it may take a while. Have a good day~

**Comment by markson14** [comment]
Since I find out timm has also implemented MobileNetV3, I am going to rewrite the entire code based on timm's version. I will close this pull request later.

**Comment by qubvel** [comment]
👍

**Comment by markson14** [comment]
I have reimplemented the code based on timm-mobilenetv3 and pass the test_model.py. Please have a check. Thank you.
image

**Comment by JulienMaille** [comment]
@markson14 I juste noticed pytorch should include Mobilenetv3 in next release, it's already commited
https://github.com/pytorch/vision/search?q=MobileNetv3&type=commits

**Comment by qubvel** [comment]
Could you merge master branch, this allow to run CI tests

**Pull request closed without merging**